controller.go

```diff
@@ -455,6 +458,44 @@ func (c *Controller) syncHandler(key string) error {
 				return err
 			}
 			id = res.ID
+		} else if c.infraConfig.Provider == "gce" {
+			provisioner, _ := provision.NewGCEProvisioner(c.infraConfig.GetAccessKey())
+
+			userData := makeUserdata(tunnel.Spec.AuthToken, c.infraConfig.UsePro(), tunnel.Spec.ServiceName)
+
+			res, err := provisioner.Provision(provision.BasicHost{
+				Name:     tunnel.Name,
+				OS:       "projects/debian-cloud/global/images/debian-9-stretch-v20191121",
+				Plan:     "f1-micro",
+				UserData: userData,
+				Additional: map[string]string{
+					"projectid": c.infraConfig.ProjectID,
+					"zone":      c.infraConfig.Zone,
+				},
+			})
+
+			if err != nil {
+				return err
+			}
+			id = res.ID
+
+			inletsPort := inletsControlPort
+
+			if c.infraConfig.UsePro() {
+				inletsPort = inletsProControlPort
+			}
+			exists, _ := provisioner.CheckInletsFirewallRuleExists(id, strconv.Itoa(inletsPort))
```

Comment:
I'd like the firewall code to be hidden inside the `Provision` call within the provision package and code.

Reply:
Sure!